specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json

```diff
@@ -0,0 +1,1318 @@
+            },
+            "workloadIdentityProviderId": {
+              "type": "string",
+              "description": "The data collection GCP workload identity provider id for this offering"
+            }
+          }
+        },
+        "auditLogsAutoProvisioningFlag": {
+          "type": "boolean",
+          "description": "Is audit logs data collection enabled"
+        },
+        "defenderAgentAutoProvisioningFlag": {
+          "type": "boolean",
+          "description": "Is Microsoft Defender for Cloud Kubernetes agent auto provisioning enabled"
+        },
+        "policyAgentAutoProvisioningFlag": {
+          "type": "boolean",
+          "description": "Is Policy Kubernetes agent auto provisioning enabled"
+        }
+      }
+    },
+    "cspmMonitorGithubOffering": {
+      "type": "object",
+      "description": "The CSPM monitoring for github offering",
+      "x-ms-discriminator-value": "CspmMonitorGithub",
+      "allOf": [
+        {
+          "$ref": "#/definitions/cloudOffering"
+        }
+      ],
+      "properties": {}
+    },
+    "cspmMonitorAzureDevOpsOffering": {
+      "type": "object",
+      "description": "The CSPM monitoring for AzureDevOps offering",
+      "x-ms-discriminator-value": "CspmMonitorAzureDevOps",
+      "allOf": [
+        {
+          "$ref": "#/definitions/cloudOffering"
+        }
+      ],
+      "properties": {}
+    },
+    "defenderCspmAwsOffering": {
+      "type": "object",
+      "description": "The CSPM P1 for Aws offering",
+      "x-ms-discriminator-value": "DefenderCspmAws",
+      "allOf": [
+        {
+          "$ref": "#/definitions/cloudOffering"
+        }
+      ],
+      "properties": {
+        "vmScanners": {
+          "type": "object",
+          "description": "The Microsoft Defender for Server VM scanning configuration",
+          "properties": {
+            "enabled": {
+              "type": "boolean",
+              "description": "Is Microsoft Defender for Server VM scanning enabled"
+            },
+            "configuration": {
+              "type": "object",
+              "description": "configuration for Microsoft Defender for Server VM scanning",
+              "properties": {
+                "cloudRoleArn": {
+                  "type": "string",
+                  "description": "The cloud role ARN in AWS for this feature"
+                },
+                "scanningMode": {
+                  "type": "string",
+                  "description": "The scanning mode for the vm scan.",
+                  "enum": [
+                    "Default"
+                  ],
+                  "x-ms-enum": {
+                    "name": "scanningMode",
+                    "modelAsString": true
+                  }
+                },
+                "exclusionTags": {
+                  "type": "object",
+                  "description": "VM tags that indicates that VM should not be scanned",
+                  "properties": {}
```

Comment:
>"properties": {} [](http://example.com/codeflow?start=18&length=16)

If its just a list of tags why not make it an array of strings?

Reply:
exclusionTags is a Key-Value pair property bag.
See how [tags](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v1/types.json#L151) are modelled as an object? same here.

Reply:
if its key value pairs of strings can u define it just like tags 
          "type": "object",
          "additionalProperties": {
            "type": "string"
          },

Reply:
also can u remove the free form object - properties pl?

Reply:
fixed